pkg/lib/cockpit-components-terminal.tsx

```diff
@@ -261,19 +351,20 @@
     }
 
     componentWillUnmount() {
-        this.disconnectChannel();
-        this.terminal.dispose();
         window.removeEventListener('resize', this.onWindowResize);
         this.onFocusOut();
+        this.unmountTerminal(this.terminal_state);
+        if (!this.props.state)
+            this.terminal_state.close();
     }
 
     setText() {
         try {
             navigator.clipboard.readText()
-                    .then(text => this.props.channel.send(text))
-                    .catch(e => this.setState({ showPastingModal: true }))
+                    .then(text => this.terminal_state.channel?.send(text))
+                    .catch(() => this.setState({ showPastingModal: true }))
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21680-336cba48-20250814-135659-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)